src/tests/ftest/util/command_utils.py

```diff
@@ -210,7 +210,11 @@ def stop(self):
                     self._command, str(state))
                 self._process.send_signal(signal_to_send)
                 if signal_list:
-                    time.sleep(5)
+                    start = time.time()
+                    while self._process._popen.poll() is None and time.time() - start < 5:
+                        time.sleep(0.01)
+                    elapsed = time.time() - start
+                    self.log.info('Waited %.2f, saved %.2f', elapsed, 5 - elapsed)
```

Comment:
can we remove logging for landing in master?

Reply:
I specifically wanted the logging so we can see what effect this has but I'm happy to remove it once we've got a run through CI to see the effect.  Johann specifically mentioned this code was running 416 times for a total of 34 minutes, when I was testing locally I saved about 80% of that, but without the logging then we lose the ability to see the effect.

Reply:
Yes, that makes sense. When Johann mentioned 80% savings I thought we have a run through CI already to collect the data (didn't know data was from local testing) and this was ready for landing if all tests pass.

Reply:
Ultimately, the savings are probably enough, and run times of green runs consistent enough that the savings will be pretty evident.  I predict at least 15 minutes savings on the VM test runs.  I only counted 241 tests each for those runs, but I wonder if some tests were doing restarts, adding up to Johann's 416.